ext/pdo_firebird/firebird_driver.c

```diff
@@ -848,7 +904,7 @@ static bool firebird_handle_set_attribute(pdo_dbh_t *dbh, zend_long attr, zval *
 						if (bval) {
 							/* turning on auto_commit with an open transaction is illegal, because
 							   we won't know what to do with it */
-							H->last_app_error = "Cannot enable auto-commit while a transaction is already open";
+							firebird_error_with_info(dbh, "HY000", 5, "Cannot enable auto-commit while a transaction is already open", 61);
```

Comment:
Using sizeof of strlen is better, the compiler will fill in the integer constant by itself